### PR TITLE
acpi: Document ACPI tables and ELF files handling

### PR DESCRIPTION
**Pull request by Zildj1an**
Document ACPI tables structs and code, including function usage examples. This is related to issue #74 titled "Document COCONUT-SVSM". Also document ELF files handling

### CODE REVIEW

**Comment by cclaudio** [comment]
When applicable, I think the two things below can be useful for people that need to work on this file:
- a comment at the top describing what spec we used in the file (e.g. Name, version)
- a comment in the structure indicating the table number in the spec it refers to

**Comment by Zildj1an** [comment]
Regarding @cclaudio comment, what do you think @joergroedel ? Should we link to any specific ACPI documentation?

**Comment by Zildj1an** [comment]
Thanks @joergroedel in that case I'm satisfied with the current state of the commit. I fixed a merge issue and that's the reason why we have a merge commit without my signature.

**Comment by Zildj1an** [comment]
I just updated this pull request, now we also document ELF files handling (see [new commit](https://github.com/coconut-svsm/svsm/pull/89/commits/5c2d2b9b5bbc07d50a692533342a6e7b31983dfc)) I will not continue documenting until this is merged to avoid bottleneck

**Comment by 00xc** [comment]
> Thanks @joergroedel in that case I'm satisfied with the current state of the commit. I fixed a merge issue and that's the reason why we have a merge commit without my signature.

I think this could be fixed by rebasing instead, no? We don't have guidelines on merges vs rebases for the project so this is fine for me, just mentioning it.

File: src/elf/mod.rs [hunk 1]
```diff
@@ -250,18 +367,39 @@ impl convert::TryFrom<(Elf64Off, Elf64Xword)> for Elf64FileRange {
...
+    ///
+    /// Returns an `ElfError` if there are issues parsing the ELF file.
+    ///
+    /// # Examples
+    ///
+    /// ```rust
+    /// use your_module::Elf64File;
+    ///
+    /// let elf_file = Elf64File::read(&elf_file_buf).unwrap();
+    /// ``
```
Comment (00xc) [inline src/elf/mod.rs:402]: There's a missing closing backtick here. `cargo doc` will warn about it:

```
warning: could not parse code block as Rust code
   --> src/elf/mod.rs:398:9
    |
398 |       /// ```rust
    |  _________^
399 | |     /// use your_module::Elf64File;
400 | |     ///
401 | |     /// let elf_file = Elf64File::read(&elf_file_buf).unwrap();
402 | |     /// ``
    | |__________^
    |
    = note: error from rustc: unknown start of token: `
    = note: `#[warn(rustdoc::invalid_rust_codeblocks)]` on by default
```

**Comment by 00xc** [comment]
The doctests for this PR are broken at the moment. Moreover, existing doctests in the main branch are also broken, so I suggest we fix those before merging these. I've opened #93 for this purpose.

You can run the doctests via `cargo test --target=x86_64-unknown-linux-gnu --doc`.

**Comment by Zildj1an** [comment]
Thanks @00xc I will reopen this PR without the issues mentioned.

**Pull request closed without merging**